src/posix/tty.rs

```diff
@@ -82,6 +83,35 @@ impl TTYPort {
         Ok(port)
     }
 
+    pub fn list_ports() -> Vec<SerialPortInfo> {
+        let context = libudev::Context::new().unwrap();
+        let mut enumerator = libudev::Enumerator::new(&context).unwrap();
+        enumerator.match_subsystem("tty").unwrap();
```

Comment:
As discussed in another comment, these errors should be handled without panicking.